src/Core/Silk.NET.Core/Native/NativeApiContainer.cs

```diff
@@ -31,18 +30,35 @@ protected NativeApiContainer(INativeContext ctx)
                     "This could be because of a SilkTouch bug, or because you're not using SilkTouch at all."
                 );
             }
-            _entryPoints = new ConcurrentDictionary<int, IntPtr>(ConcurrencyLevel, slotCount);
-            GcUtility = new GcUtility(ConcurrencyLevel, slotCount);
+            if (!tableInitialized)
+                _vTable.Init(_ctx, slotCount);
+            GcUtility = new GcUtility(1, CoreGcSlotCount());
+            // ReSharper restore VirtualMemberCallInConstructor
         }
 
+        protected NativeApiContainer(INativeContext ctx) : this(ctx, new ConcurrentDictionaryVTable(), false)
+        { }
+
         public GcUtility GcUtility { get; }
 
+        public IVTable CurrentVTable => _vTable;
+
         public void Dispose()
         {
             _ctx.Dispose();
+            CurrentVTable.Dispose();
         }
 
         protected virtual int CoreGetSlotCount() => 0;
+        protected virtual int CoreGcSlotCount() => 0;
```

Comment:
Are these values ever different?

Reply:
The root values are always zero, otherwise they are only the same if every method slot also requires a GC Slot (Pins something via GCUtil)

Reply:
Ahhh right ok.